docs/tutorials/testing.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: page
+title: Automated Testing
+date: 2017-07-03 15:33:06
+author: Louis Pilfold
+permalink: /tutorials/testing
+---
+
+Thanks to Haskell's type system we can avoid most the errors that we might
+encounter when using less safe languages, but some automated tests can still
+help verify our code is correct. Let's explore how we can test our application
+through the Spock based HTTP interface.
+
+## Hello world
+
+```haskell
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+import Web.Spock
+import Web.Spock.Config
+
+main :: IO ()
+main =
+    do spockCfg <- defaultSpockCfg () PCNoDatabase ()
+       runSpock 8080 (spock spockCfg app)
+
+app :: SpockM () () () ()
+app =
+    do get root $
+           text "Hello World!"
+       get ("hello" <//> var) $ \name ->
+           do text ("Hello " <> name)
+```
+
+Here is a simple Spock application, similar to the one from the
+[Hello World][hw-tut]. It has two routes, `/` which responds with "Hello,
+world!", and `/hello/:name`, which responds with "Hello `$NAME`", where name
+is the value of the second path segment.
+
+It type checks, so we are confident that the application will run without
+crashing, but we might want some tests to check that the business logic has
+been correctly implemented. We can do this with a little help from the
+[Hspec][hspec] and [Hspec-Wai][hspec-wai] libraries.
+
+Add Hspec and Hspec-Wai to your tests dependencies in your project's
+`package.yaml` file or Cabal file. If you're using `package.yaml` it may look
```

Comment:
I'm afraid I'm new to Haskell and am not very familiar with Cabalfiles, how does this look?